modulus/datapipes/climate/climate.py

```diff
@@ -0,0 +1,829 @@
+        num_workers: int = 1,  # TODO: is there a faster good default?
+        device: Union[str, torch.device] = "cuda",
+        process_rank: int = 0,
+        world_size: int = 1,
+    ):
+        super().__init__(meta=MetaData())
+        self.sources = list(sources)
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.shuffle = shuffle
+        self.dt = dt
+        self.start_year = start_year
+        self.lsm_filename = lsm_filename
+        self.lsm_variable = lsm_variable
+        self.geopotential_filename = geopotential_filename
+        self.geopotential_variable = geopotential_variable
+        self.use_latlon = use_latlon
+        self.latlon_bounds = latlon_bounds
+        self.process_rank = process_rank
+        self.world_size = world_size
+        if isinstance(patch_size, int):
+            patch_size = (patch_size, patch_size)
+        self.patch_size = patch_size
+        self.num_samples_per_year = num_samples_per_year
+        self.logger = PythonLogger()
+
+        # Determine outputs of pipeline
+        self.pipe_outputs = []
+        for (i, spec) in enumerate(self.sources):
+            name = spec.name if spec.name is not None else i
+            self.pipe_outputs += [f"state_seq-{name}", f"timestamps-{name}"]
+            if spec.use_cos_zenith:
+                self.pipe_outputs.append(f"cos_zenith-{name}")
+        if self.lsm_filename is not None:
+            self.pipe_outputs.append("land_sea_mask")
+        if self.geopotential_filename is not None:
+            self.pipe_outputs.append("geopotential")
+        if self.use_latlon:
+            self.pipe_outputs.append("latlon")
+            self.pipe_outputs.append("cos_latlon")
+
+        # Set up device, needed for pipeline
+        if isinstance(device, str):
+            device = torch.device(device)
+
+        # Need a index id if cuda
+        if device.type == "cuda" and device.index is None:
+            device = torch.device("cuda:0")
+        self.device = device
+
+        # Load all data files and statistics
+        for spec in sources:
+            spec.parse_dataset_files(
+                num_samples_per_year=num_samples_per_year, patch_size=patch_size
+            )
+        for (i, spec_i) in enumerate(sources):
+            for spec_j in sources[i + 1 :]:
+                if not spec_i.dimensions_compatible(spec_j):
+                    raise ValueError("Incompatible data sources")
+        if self.lsm_filename is not None:
+            self._load_land_sea_mask()
+        if self.geopotential_filename is not None:
+            self._load_geopotential()
+        if self.use_latlon or any(spec.use_cos_zenith for spec in sources):
+            self._load_latlon()
+
+        # Create pipeline
+        self.pipe = self._create_pipeline()
+
+    def _read_invariant_autoflip(self, filename, var, normalize=False):
+        """Get invariant from netCDF file."""
+        with nc.Dataset(filename) as ds:
```

Comment:
Can we replace netcdf with the more flexible xarray load? That way we can move towards supporting other file formats.

Reply:
Yes we can certainly change these to xarray if preferred.

Reply:
Changed to xarray in 45661fe98135162d952302ce9e7deea2b1da70ae